include/revng/Model/TupleTree.h

```diff
@@ -997,6 +1013,16 @@ class TupleTreeReference {
     return fromPath(*stringAsPath<RootT>(Path));
   }
 
+  std::weak_ordering operator<=>(const TupleTreeReference &Other) const {
+    // The paths are the same even if they are referred to different roots
+    return Path <=> Other.Path;
+  }
+
+  bool operator==(const TupleTreeReference &Other) const {
+    // The paths are the same even if they are referred to different roots
+    return Path == Other.Path;
+  }
```

Comment:
Isn't this superfluous given `operator<=>`?

Reply:
Nice catch. The point here is that I started adding `operator<=>` and `operator==` during development because I needed them to get the first implementation through. So, when I added `operator<=>`, `operator==` was already there, and in the rush I did not remove it. Also, notice that `operator<=>` returns `weak_ordering` here, which is why removing `operator==` would not be possible, because AFAIU `operator==` cannot be synthesized starting from `operator<=>` when the latter return `weak_ordering`. However, because of how the two operators are defined  here, the sane thing to do would have been to only leave `operator<=>` but changing it to return `strong_ordering`, so that `operator==` could be synthesized from it.

Anyway, in the final implementation `operator<=>` is not needed anymore, at least for now. So I dropped it entirely. Now we only have `operator==`.